src/DataDog.Tracing.Sql/TraceDbCommand.cs

```diff
@@ -1,51 +1,105 @@
 ﻿using System;
 using System.Data;
+using System.Data.Common;
+using System.Threading;
+using System.Threading.Tasks;
 
 namespace DataDog.Tracing.Sql
 {
-    public class TraceDbCommand : IDbCommand
+    public class TraceDbCommand : DbCommand
```

Comment:
Using the base classes rather than the interfaces is going to break downstream consumers.

Reply:
`IDbCommand` and `IDbConnection` are pre .NET 2.0 interfaces, when async/await wasn't still a thing, and therefore lack support for asynchronous methods.

`DbCommand` and `DbConnection` were introduced after async/await support, and to avoid breaking previous implementations by adding new functionality Microsoft decided to create those abstract base classes to extend further the interfaces with async methods.

All connectors (MySql connector, SQLServer connector, PostgreSQL Npgsql connector, etc...) implement the abstract classes and not the interfaces nowadays, so this change shouldn't break any consumer (other than those using old .NET versions, but those don't support either .NET Standard so it doesn't even matter because they wouldn't be able to consume this library anyways).

Reply:
It certainly breaks the one consumer I know about since they use the interfaces :)

Reply:
Do you intend to do the work of fixing this in WB?

Reply:
Yes, I'll gladly work in the `RepositoryBase` implementation of the async overloads and all the Repositories that inherit from it!

It's probably a lot of work but I think it's worth the effort. Also we can leave the synchronous methods for backwards compatibility, maybe with the `[ObsoleteAttribute]`.

Reply:
Here's another `DbCommand` version: https://github.com/Informatievlaanderen/datadog-tracing/blob/master/src/Be.Vlaanderen.Basisregisters.DataDog.Tracing.Sql/EntityFrameworkCore/TraceDbCommand.cs#L8 :)